src/sctools/bam.py

```diff
@@ -441,6 +441,42 @@ def get_tag_or_default(alignment: pysam.AlignedSegment, tag_key: str, default: O
         return default
 
 
+def sort_by_tags_and_queryname(
+        records: Iterable[pysam.AlignedSegment],
+        tags: Iterable[str] = None) -> Iterable[pysam.AlignedSegment]:
```

Comment:
I think this type is probably `Optional[Iterable[str]]` since it's assigned `None`.